sunpy/net/dataretriever/client.py

```diff
@@ -206,17 +206,26 @@ def get(self, qres, **kwargs):
         -------
         Results Object
         """
+        details = (qres.__dict__['client']).map_
+
         urls = []
         for qrblock in qres:
             urls.append(qrblock.url)
 
+        filenames = []
+        for url in urls:
+            filenames.append(url.split('/')[-1])
+
         res = Results(lambda x: None, 0, lambda map_: self._link(map_))
 
         dobj = Downloader(max_conn=len(urls), max_total=len(urls))
-        for aurl, ncall in list(zip(urls, map(lambda x: res.require([x]),
-                                              urls))):
-            dobj.download(aurl, kwargs.get('Path', None), ncall,
-                          kwargs.get('ErrorBack', None))
+
+        for aurl, ncall, filename in list(zip(urls, map(lambda x: res.require([x]),
+                                              urls), filenames)):
+            temp_dict = details.copy()
+            temp_dict['file'] = filename
+            dobj.download(aurl, kwargs.get('path', None), ncall,
```

Comment:
these kwargs should be explicit in the calling signature (you should also leave `**kwargs` there).